lighthouse-core/gather/gatherers/anchor-elements.js

```diff
@@ -87,8 +114,25 @@ class AnchorElements extends Gatherer {
       return (${collectAnchorElements})();
     })()`;
 
-    /** @type {Array<LH.Artifacts.AnchorElement>} */
-    return driver.evaluateAsync(expression, {useIsolation: true});
+    /** @type {LH.Artifacts['AnchorElements']} */
+    const anchors = await driver.evaluateAsync(expression, {useIsolation: true});
+    await driver.sendCommand('DOM.enable');
+
+    // DOM.getDocument is necessary for pushNodesByBackendIdsToFrontend to properly retrieve nodeIds if the `DOM` domain was enabled before this gatherer, invoke it to be safe.
+    await driver.sendCommand('DOM.getDocument', {depth: -1, pierce: true});
+    const anchorsWithEventListeners = anchors.map(async anchor => {
+      const eventListeners = await getEventListeners(driver, anchor.devtoolsNodePath);
+      const hasClickHandler = eventListeners.some(({type}) => type === 'click');
```

Comment:
I feel this artifact doesn't match completely the architectural intention for what an artifact is, but I think we agree that returning something like `hasClickHandler` is an okay compromise considering the large event handler payloads otherwise.

That being said if you want to change the name of `hasClickHandler` or even, return something more meaningful (like an array of _all_ listeners, _minus_ the large handler string, and then have the audit filter for what it needs), that's ok. I did this as a comprimise and because I don't see it being used by anything else.

Reply:
I'm not a huge fan of `hasClickHandler` for the intention of a pure artifact purpose like you mention.

How would you feel about going back to the `listeners` property but only exposing the `listener.type` on each for now? i.e. `anchor.listeners = eventListeners.map(({type}) => ({type}))`

That should minimize the impact on artifact size, preserve the purity of the artifact intention, and leave the door open to bring in the other properties in the future should this or other audits require it.